homeassistant/components/weatherflow_cloud/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for WeatherflowCloud integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientResponseError
+import voluptuous as vol
+from weatherflow4py.api import WeatherFlowRestAPI
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+async def _async_validate_api_token(api_token) -> bool:
+    """Validate the API token."""
+
+    async with WeatherFlowRestAPI(api_token) as api:
+        await api.async_get_stations()
+
+    return True
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlowCloud."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        if user_input is None:
+            return await self._show_setup_form(user_input)
+
+        await self.async_set_unique_id(user_input[CONF_API_TOKEN])
```

Comment:
Can the api token change? Can you revoke it? If so, this isn't a good unique_id

Reply:
The token doesn't ever change - but it can be revoked.

- A token maps to a "set" of devices.
- If you add more devices to your account then they would be attached to existing tokens...
- There doesn't seem to be any other way to uniquely identify a set of devices... 
- A token can be deleted but they never expire

My thought process was I don't want to allow a user to enter the same token twice, but if say I had multipel tokens (my house, my rental, my parents) I'd want to allow for 3 entries into HA.

Any thoughts on alternatives?

Reply:
Use `_async_abort_entries_match()` instead and make a reauth flow so the token can be fixed if neccessary.

Reply:
Agreed

Reply:
Like so or do I nix the `self._abort_if_unique_id_configured()`


![image](https://github.com/home-assistant/core/assets/6491743/78efd421-d713-4d72-b482-8ac3b021c3af)

Reply:
Remove both unique_id lines

Reply:
And add pass user_input to the `self._async_abort_entries_match()`